packages/sdk/src/swap/sdk.ts

```diff
@@ -350,6 +350,32 @@ export class SwapSDK {
     if (!result.success) throw new Error(result.reason);
   }
 
+  async approveAndExecuteSwap(
+    params: ExecuteSwapParams,
+    txOpts: TransactionOptions & { signer?: Signer } = {},
```

Comment:
should we accept nonces or maybe a function to get the next nonce?

Reply:
As agreed, ignoring nonces for now.